weaver/cli.py

```diff
@@ -242,6 +242,15 @@ class RequestAuthHandler(AuthHandler, HTTPBasicAuth):
     """
     Base class to send a request in order to retrieve an authorization token.
     """
+
+    def __init__(self, token=None, **kwargs):
+        # type: (Any, **Any) -> None
+        if token is None:
+            self.token = token
+        else:
+            self.token = self.parse_token(token)
+        AuthHandler.__init__(self, **kwargs)
```

Comment:
Might be better to switch the order. For example in the case of HTTP Basic auth, if the token was provided, the username and password could be inferred from the value.

Reply:
Instead of having two calls to `parse_token` (if I implement the suggestion here: https://github.com/crim-ca/weaver/pull/599#discussion_r1483732682), I could just call `parse_token` once in `__call__` for both cases (if the token is provided or if we get if from a request)

Reply:
Yes. That's what I had in mind as well. 👍